core/trino-main/src/main/java/io/trino/server/security/oauth2/OidcDiscovery.java

```diff
@@ -114,6 +115,7 @@ private OAuth2ServerConfig readConfiguration(String body)
             else {
                 userinfoEndpoint = Optional.empty();
             }
+            Optional<URI> endSessionEndpoint = Optional.of(getRequiredField("end_session_endpoint", metadata.getEndSessionEndpointURI(), END_SESSION_URL, Optional.empty()));
```

Comment:
Should provide an override value for END_SESSION_URL which comes from the OidcDiscoveryConfig. Similar to the other URL values. This allows non-standard endpoint values.

Reply:
That's a good point but `OidcDiscoveryConfig` actually allows providing urls explicitly (effectively overriding those obtained from metadata discovery) for backward compatibility, allowing for non-standard endpoint values wasn't the objective. We intend to drop these properties https://github.com/trinodb/trino/pull/18101
That said, is it something you think we should support? Is this something you use these properties for?

Reply:
This _END_SESSION_URL, Optional.empty())_ becomes this _END_SESSION_URL, logoutUrl)_, where logoutUrl is set in the constructor

logoutUrl = requireNonNull(oidcConfig.getEndSessionUrl(), "logoutUrl is null");

I had to make these changes for a customer, who is using Ping. The "standard" endpoint found with _end_session_endpoint_ is not being found because the well-known endpoint is returning _ping_end_session_endpoint_. So the customer had to be able to specify the logout URL explicitly.

Reply:
But if we disable the OIDC discovery then we could inject them via `StaticConfigurationProvider` - we could provide custom end-points right ? cc : @lukasz-walkiewicz Are we removing that endpoint also ?

Reply:
No, `StaticConfigurationProvider` is here to stay. I think we ultimately would like to remove this overriding some day and users should provide configuration explicitly if the IdP does not conform OIDC spec.

Reply:
@thorbjornsen In this case we could still use `StaticOAuth2ServerConfiguration` but we need to specify the configuration explicitly.